src/builder/build_traits.rs

```diff
@@ -641,11 +640,11 @@ fn wire_up<T: Dataflow + ?Sized>(
 /// Return None if Const kind
 /// Panics if port not valid for Op or port is not Const/Value
 fn check_classical_value(
-    base: &HugrMut,
+    base: &impl AsRef<Hugr>,
```

Comment:
This could probably also be `&impl HugrMut`, but `AsRef<Hugr>` seems more general (? or is it)

Reply:
At present `HugrMut` has `AsRef<Hugr>` as a prereq, so that'd be strictly less general - some `AsRef<Hugr>` things don't have the HugrMut methods (but probably could have if we wanted). However, in the future I hope that HugrMut will not require being able to retrieve an actual Hugr (even a reference), rather, HugrMut will be able to return an immutable *view* onto what might be only part of a Hugr. So if we wanted to plan ahead for that, I should change this here to HugrMut, but that will be another PR(-series)...